<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>lpc_flash_app</class>
 <widget class="QMainWindow" name="lpc_flash_app">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>557</width>
    <height>353</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>lpc_flash_app</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QComboBox" name="comboBox_mcu">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>70</y>
      <width>85</width>
      <height>27</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>lpc1315</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>lpc1316</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>lpc1317</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>lpc1345</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>lpc1346</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>lpc1347</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>50</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>MCU:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Port:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_port_name">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>20</y>
      <width>113</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>0</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Baud:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_baud">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>20</y>
      <width>81</width>
      <height>27</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>2400</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>4800</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>9600</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>19200</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>38400</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>57600</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>115200</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>0</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Data bits:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_data_bits">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>20</y>
      <width>61</width>
      <height>27</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>5</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>6</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>7</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>8</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>50</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Parity:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_parity">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>70</y>
      <width>71</width>
      <height>27</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>None</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Odd</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Even</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Space</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Mark</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>50</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Stop bits:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_stop_bits">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>70</y>
      <width>61</width>
      <height>27</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>1</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>1.5</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>2</string>
     </property>
    </item>
   </widget>
   <widget class="QToolButton" name="browse_file">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>130</y>
      <width>24</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>...</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>110</y>
      <width>331</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Select binary file to program to microcontroller:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_file_path">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>130</y>
      <width>341</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="download_button">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>170</y>
      <width>81</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Download</string>
    </property>
   </widget>
   <widget class="QPushButton" name="erase_button">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>170</y>
      <width>99</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Erase</string>
    </property>
   </widget>
   <widget class="QPushButton" name="connect_button">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>20</y>
      <width>91</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Connect</string>
    </property>
   </widget>
   <widget class="QPushButton" name="disconnect_button">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>70</y>
      <width>91</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Disconnect</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_xtal_freq">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>20</y>
      <width>91</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>0</y>
      <width>111</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>MCU XTAL, kHz</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_reset">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>170</y>
      <width>99</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Reset</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="checkBox_valid_code">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>100</y>
      <width>98</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Vector 7</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>90</y>
      <width>81</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Valid Code</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="checkBox_verify">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>140</y>
      <width>98</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Verify</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_file_path_upload">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>220</y>
      <width>341</width>
      <height>27</height>
     </rect>
    </property>
   </widget>
   <widget class="QToolButton" name="toolButton_browse_file_upload">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>215</y>
      <width>24</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>...</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_upload">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>250</y>
      <width>99</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Upload</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>200</y>
      <width>391</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Select path and file name to read from microcontroller:</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>557</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionInfo"/>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionLoad_configuration_2">
   <property name="text">
    <string>Load configuration</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Load</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionInfo">
   <property name="text">
    <string>Info</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
